lib/dotcom/trip_plan/itinerary_groups.ex

```diff
@@ -69,22 +55,43 @@ defmodule Dotcom.TripPlan.ItineraryGroups do
     grouped_itineraries
   end
 
-  defp to_summarized_group(grouped_itineraries) do
-    %{
-      itineraries: ItineraryTag.sort_tagged(grouped_itineraries),
-      summary: summary(grouped_itineraries)
-    }
+  defp limit_itinerary_count(itineraries, take_from_end?) do
```

Comment:
Not sure how this suggestion will play with [this other comment above](https://github.com/mbta/dotcom/pull/2329/files#r1919161002), but this could be accomplished with pattern matching

```elixir
defp limit_itinerary_count(itineraries, true), do: Enum.take(itineraries, -@max_per_group)
defp limit_itinerary_count(itineraries, false), do: Enum.take(itineraries, @max_per_group)
```

Or similarly
```elixir
defp limit_itinerary_count(itineraries, %{take_from_end: true}), do: Enum.take(itineraries, -@max_per_group)
defp limit_itinerary_count(itineraries, _), do: Enum.take(itineraries, @max_per_group)
```